core/Engine/Tweek.Engine.Drivers/Rules/RulesRepository.cs

```diff
@@ -26,33 +26,50 @@ public class RulesRepository : IRulesRepository, IDisposable
                 .Do(_ => LastCheckTime = scheduler.Now.UtcDateTime)
                 .DistinctUntilChanged()
                 .Do(version => logger.LogInformation($"Detected new rules version: {version}"))
-                .Select(version => Observable.FromAsync(ct => rulesDriver.GetRuleset(version, ct)).Do(_ => CurrentLabel = version))
+                .Select(version => Observable.FromAsync(ct => rulesDriver.GetRuleset(version, ct))
+                .Select(rules=> (version, rules)))
```

Comment:
`Select` is chained to `FromAsync` and not to previous `Select`
Indent it for readability...